packages/driver/src/cy/multi-domain/logs_manager.ts

```diff
@@ -60,6 +63,19 @@ export class LogsManager {
     const isEnded = log.get('ended')
 
     if (isEnded) {
+      if (log.get('state') === 'failed') {
+        const err = log.get('err')
+        let parsedError = correctStackForCrossDomainError(err, this.userInvocationStack)
+
+        // The toJSON method on the Cypress.Log converts the 'error' property to 'err', so when the log gets
+        // serialized from the SD to the PD, we need to do the opposite
+        log.set('error', parsedError)
+        if ((logAttrs.consoleProps as any)?.Error) {
+          // Update consoleProps for when users pin failed commands in the reporter so correct error messages are displayed in the console
+          (logAttrs.consoleProps as any).Error = parsedError.stack
+        }
+      }
+
       delete this.logs[attrs.id]
       deferred.resolve()
```

Comment:
Can also reduce nesting a bit here with early returns.

```js
if (!isEnded) return

// ... the rest as-is ...
```

Also, how about the following?

```js
const err = correctStackForCrossDomainError(log.get('err'), this.userInvocationStack)

// rest uses `err` instead of `parsedError`
```